File: src/client/src/types/terminals/terminalView.ts
import { Aspect } from "../common/aspect";
import { AttributeTypeReferenceView } from "../common/attributeTypeReferenceView";
import { ImfType } from "../common/imfType";
import { RdlClassifier } from "../common/rdlClassifier";
import { RdlPurpose } from "../common/rdlPurpose";
import { Direction } from "./direction";
import { RdlMedium } from "./rdlMedium";

export interface TerminalView extends ImfType {
  classifiers: RdlClassifier[];
  purpose: RdlPurpose | null;
  notation: string | null;
  aspect: Aspect | null;
  medium: RdlMedium | null;
  qualifier: Direction;
  attributes: AttributeTypeReferenceView[];
}

File: src/server/Tyle.Api/Terminals/TerminalView.cs
using Tyle.Api.Common;
using Tyle.Core.Common;
using Tyle.Core.Terminals;

namespace Tyle.Api.Terminals;

public class TerminalView : ImfType
{
    public ICollection<RdlClassifier> Classifiers { get; set; } = new List<RdlClassifier>();
    public RdlPurpose? Purpose { get; set; }
    public string? Notation { get; set; }
    public string? Symbol { get; set; }
    public Aspect? Aspect { get; set; }
    public RdlMedium? Medium { get; set; }
    public Direction Qualifier { get; set; }
    public ICollection<AttributeTypeReferenceView> Attributes { get; set; } = new List<AttributeTypeReferenceView>();
    public string Kind => nameof(TerminalView);
}